operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ConfigParameter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.operator.resource;
+
+import io.strimzi.operator.common.InvalidConfigurationException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Models a configuration parameter, identified by a unique key, which may be required, and if not may have a default value.
+ * Optional parameters without a default value implicitly have a null default.
+ * The key is also the name of the environment variable from which the value may be read, when it is read from the environment.
+ *
+ * @param key Configuration parameter name/key
+ * @param <T> Type of object
+ * @param type type of the default value
+ * @param defaultValue default value of the configuration parameter
+ * @param required  If the value is required or not
```

Comment:
The `map` field seems to be missing here?

Tip: if you align the descriptions with tabs, the JavaDocs will be much more readable.